daemon/stats.go

```diff
@@ -43,11 +43,11 @@ func (daemon *Daemon) ContainerStats(ctx context.Context, prefixOrName string, c
 
 	var preCPUStats types.CPUStats
 	getStatJSON := func(v interface{}) *types.StatsJSON {
-		ss := v.(*types.StatsJSON)
+		ss := *v.(*types.StatsJSON)
```

Comment:
The * operation is still racy. I think the safest way is to publish values to pubsub.